### ISSUE TITLE
Access Requests to MacOS Document/Desktop Folders

### ISSUE THREAD

**Issue by dcrawl** [godotengine/godot#61251]
### Godot version

4.0 dev alpha 8

### System information

MacOS 12.3.1 Arm64 

### Issue description

My Godot 4.app has full access to my drive in my Security and Privacy settings.  Yet if my Godot project is located in my Desktop or Documents folder, when I start Godot, it prompts me for access to the folder.  When I run the project, it prompts me multiple times before it will start.  Even if I choose OK, it continues to prompt me every time.

If I move the project to a normal folder location, then it works fine.

### Steps to reproduce

Create a new project in your Documents folder.
Save the new scene with Node2D as the root.
Run the scene.

**Comment by Calinou** [comment]
cc @bruvzg

**Comment by dcrawl** [comment]
No.  I've never had this problem on 3.4.4 or 3.5 but then again I've never compiled Godot before.  I will give it a try.

J

**Comment by bruvzg** [comment]
Try removing it from the Privacy Settings (with a `-` button), this is not specific to Godot, and sometimes happens when you update the app or have multiple copies.

**Comment by dcrawl** [comment]
My Godot 4 app is named "Godot 4.app" so it shows as different from my Godot 3.x in the security list.  I removed all references to all versions of Godot in the Full Disk and the Folders permissions.    But it keeps asking me multiple times.  And I verified that the Godot 4.app is listed in the Folders security for the Desktop.   So the request for permission was stored but it doesn't seem to be verifying against it properly.

When I compiled and bundled the app, I followed the compile instructions except that the final universal file name was different than the one listed in the instructions.  Other than that, the compile of Godot 4 was uneventful.
